### ISSUE TITLE
No string to array assignment inspection?

### ISSUE THREAD

**Issue by iakoder** [kalessil/phpinspectionsea#918]
Just logged this https://youtrack.jetbrains.com/issue/WI-42120 and thought you might be interested.  I'm surprised that no one has detected this yet.

```
<?php

class Questionable {

    /**
     * Why isn't this flagged by any inspection as an error?
     *
     * @var array $arr
     */
    private $arr = '';

    public function throwsException() {
        $this->arr[] = "Throws exception in PHP 7.1";
    }

}

$q = new Questionable();
$q->throwsException();
echo "PHP 7.1 won't output this.";
```

**Comment by kalessil** [comment]
That's as usually with PhpDoc ¯\_(ツ)_/¯. Similar cases applying to method/function parameters.
In this project I declining PhpDoc related tickets (just for the focus sense), but it worth checking id PhpStan catches this (I can imagine they could implement this if not yet catching).

PS: voted for the issue.
